kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java

```diff
@@ -292,9 +295,9 @@ private void doPutAssets(final KuraRequestPayload reqPayload, final KuraResponse
             boolean flag = true;
             try {
                 this.wrapValue(assetRecord, userValue, userType);
-            } catch (final NumberFormatException nfe) {
+            } catch (final IllegalArgumentException e) {
```

Comment:
Please have a look at the javadoc https://docs.oracle.com/javase/7/docs/api/java/lang/NumberFormatException.html